lib/spack/docs/build_settings.rst

```diff
@@ -179,15 +179,72 @@ Specific limitations include:
 
 * A package must define ``executables`` and ``determine_spec_details``
   for Spack to locate instances of that package.
-* This is currently intended to find build dependencies rather than
-  library packages.
+* This implementation only collects and examines executable files, so it
+  is only useful for build/run dependencies.
```

Comment:
*may only be useful

If only because we actually *could* probably build MPI detectors with just this functionality -- the compiler wrappers can tell us a lot.